src/io/parquet/reader_async.rs

```diff
@@ -0,0 +1,44 @@
+use crate::error::Result;
+use crate::io::parquet::geoparquet_metadata::build_arrow_schema;
+use crate::io::parquet::reader::GeoParquetReaderOptions;
+use crate::table::GeoTable;
+
+use futures::stream::TryStreamExt;
+use parquet::arrow::async_reader::{AsyncFileReader, ParquetRecordBatchStreamBuilder};
+
+pub async fn read_geoparquet_async<R: AsyncFileReader + Unpin + Send + 'static>(
+    reader: R,
+    options: GeoParquetReaderOptions,
+) -> Result<GeoTable> {
+    let builder = ParquetRecordBatchStreamBuilder::new(reader)
+        .await?
+        .with_batch_size(options.batch_size);
+
+    let (arrow_schema, geometry_column_index, target_geo_data_type) =
+        build_arrow_schema(&builder, &options.coord_type);
+
+    let stream = builder.build()?;
+    let batches = stream.try_collect::<_>().await?;
+
+    GeoTable::from_arrow(
+        batches,
+        arrow_schema,
+        Some(geometry_column_index),
+        target_geo_data_type,
+    )
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use tokio::fs::File;
+
+    #[tokio::test]
+    async fn nybb() {
+        let file = File::open("fixtures/geoparquet/nybb.parquet")
+            .await
+            .unwrap();
+        let options = GeoParquetReaderOptions::new(65536, Default::default());
+        let _output_ipc = read_geoparquet_async(file, options).await.unwrap();
```

Comment:
Note this isn't IPC; it's Arrow memory directly. IPC is "Arrow in a single contiguous buffer that can be sent to a different program"

Reply:
Was wondering what IPC meant :laughing: I've renamed it to `_output_geotable` in 93cad4cad1f331b13cc107671882e1937e49cdae.